build.js

```diff
@@ -0,0 +1,108 @@
+const execSync = require('child_process').execSync;
+const path = process.argv[2];
+const webcomponent = process.argv[3];
+
+class WebComponentBuild {
+	constructor(path, webcomponent) {
+		this.path = path;
+		this.webcomponent = webcomponent;
+	}
+
+	execute() {
+		this.__buildEs6();
+		this.__buildEs6Min();
+		this.__buildNode();
+		this.__commit();
+	}
+
+	__buildEs6() {
+		const es6BuildFile = `${this.path}/vl-${this.webcomponent}.js`;
+		this.__copySrcTo(es6BuildFile);
+		this.__maakStyleImportAbsoluut(es6BuildFile);
+		this.__maakDistImportsAbsoluut(es6BuildFile);
+		this.__maakSrcImportsAbsoluut(es6BuildFile);
+	}
+	
+	__buildEs6Min() {
+		const es6MinBuildFile = `${this.path}/vl-${this.webcomponent}.min.js`;
+		this.__copySrcTo(es6MinBuildFile);
+		this.__maakDistImportsAbsoluut(es6MinBuildFile);
+		this.__maakSrcImportsAbsoluut(es6MinBuildFile);
+		this.__inlineCss(es6MinBuildFile, this.path);
+		this.__minify(es6MinBuildFile);
+	}
+
+	__buildNode() {
+		const nodeBuildFile = `${this.path}/vl-${this.webcomponent}.src.js`;
+		this.__copySrcTo(nodeBuildFile);
+		this.__vervangWebcomponentenImportsDoorCommonJsImports(nodeBuildFile);
+		this.__vervangThidPartyImportsDoorCommonJsImports(nodeBuildFile);
+		this.__vervangLocalDistImportsDoorCommonJsImports(nodeBuildFile);
+		this.__vervangLocalSrcImportsDoorCommonJsImports(nodeBuildFile);
+		this.__inlineCss(nodeBuildFile, this.path);
+	}
+	
+	__copySrcTo(destFile) {
+		executeScript(`cp -R -f ${this.path}/src/*.src.js ${destFile}`);
```

Comment:
Valt denk ik ook gemakkelijk in javascript te doen zonder dat we een extern commando aan moeten roepen. Bonus is dat het platform onafhankelijk is dan.

Reply:
pak ik mee in het volgende ticketje waar we meerdere files moeten kopieren. (UIG-510)